tuf/repository_tool.py

```diff
@@ -1809,32 +1810,32 @@ def add_restricted_paths(self, restricted_paths, child_rolename):
       raise securesystemslib.exceptions.Error(repr(child_rolename) + ' does'
         ' not exist.')
 
-    for restricted_path in restricted_paths:
-      # Do the restricted paths fall under the repository's targets directory?
+    for path in paths:
+      # Do the delegated paths fall under the repository's targets directory?
       # Append a trailing path separator with os.path.join(path, '').
       targets_directory = os.path.join(self._targets_directory, '')
-      if not restricted_path.startswith(targets_directory):
-        logger.debug(repr(restricted_path) + ' does not live under the'
+      if not path.startswith(targets_directory):
+        logger.debug(repr(path) + ' does not live under the'
             ' repository\'s targets'
           ' directory: ' + repr(self._targets_directory))
 
-      relative_paths.append(restricted_path[len(self._targets_directory):])
+      relative_paths.append(path[len(self._targets_directory):])
 
     # Get the current role's roleinfo, so that its delegations field can be
     # updated.
     roleinfo = tuf.roledb.get_roleinfo(self._rolename, self._repository_name)
 
-    # Update the restricted paths of 'child_rolename' to add relative paths.
+    # Update the delegated paths of 'child_rolename' to add relative paths.
     for role in roleinfo['delegations']['roles']:
       if role['name'] == child_rolename:
-        restricted_paths = role['paths']
+        delegated_paths = role['paths']
 
     for relative_path in relative_paths:
-      if relative_path not in restricted_paths:
-        restricted_paths.append(relative_path)
+      if relative_path not in delegated_paths:
+        delegated_paths.append(relative_path)
```

Comment:
Could it happen that `delegated_paths` is referenced here before assignment?

Reply:
Also, the only purpose of above two loops seems to be creating the `delegated_paths` list, which isn't used anywhere.